cdk-testsuite/tests/openshiftSanity.py

```diff
@@ -0,0 +1,269 @@
+        projectname (string): name of the project to be added to the openshift server
+        registry (string): registry path/location
+        servicename (string): name of the service to be exposed outside
+        tempalte (boolean): if creating the s2i using openshift template then it takes value as True otherwise the default value False 
+        dbservicename (string): Takes the database service name if specified otherwise default
+    '''
+    output = openshift.oc_usr_login(self, port, username, password)
+    self.assertIn("Login successful", output, "Login failed")
+        
+    output = openshift.add_new_project(self, projectname)
+    self.assertIn(projectname, output, "Failed to create " +projectname)
+    
+    if not tempalte:
+        output = openshift.add_new_app(self, registry)
+        partenLst = []
+        lst = registry.split("/")
+        repo = lst[len(lst) - 1]
+        for lines in output.splitlines():
+            parten = re.search(r"^(?=.*?\b\\*\b)(?=.*?\bfailed\b)(?=.*?\b%s\b).*$" %repo, lines)
+            partenLst.append(parten)
+        match = "NotFound"
+        for i in partenLst:
+            if i != None:
+                match = "Found"
+                break
+        self.assertIn("NotFound", match, registry +" deployment failed")
+    else:
+        output = openshift.add_new_template(self, registry)
+        partenLst = []
+        for lines in output.splitlines():
+            parten = re.search(r"^(?=.*?\b\\*\b)(?=.*?\bfailed\b)(?=.*?\b%s\b).*$" %registry, lines)
+            partenLst.append(parten)
+        match = "NotFound"
+        for i in partenLst:
+            if i != None:
+                match = "Found"
+                break
+        self.assertIn("NotFound", match, registry +" deployment failed")
+        if "default" not in dbservicename:
+            del partenLst[:]
+            for lines in output.splitlines():
+                parten = re.search(r"^(?=.*?\bdeploys\b)(?=.*?\b%s\b)(?=.*?\bopenshift/%s\b).*$" %(dbservicename, dbservicename), lines)
+                partenLst.append(parten)
+            match = "NotFound"
+            for i in partenLst:
+                if i != None:
+                    match = "Found"
+                    break
+            self.assertIn("Found", match, dbservicename +" deployment failed")
+            
+    if not tempalte:    
+        output = openshift.oc_port_expose(self, servicename)
+        self.assertIn("exposed", output, "Service failed to expose " +projectname)
+    else:
+        pass
+                                
+    output = openshift.routing_cdk(self, servicename, projectname)
+    self.assertIn("HTTP/1.1 200 OK", output, "Service " +servicename +"-" +projectname +" fail to expose to outside")
+                                    
+    output = openshift.oc_get_pod(self)
+    self.assertIn("Running", output, "Failed to run pod")
+                                        
+    output = openshift.oc_delete(self, projectname)
```

Comment:
Here we can not use tearDown() because of parameter passed in each test script will not be handled through tearDown().